src/ansys/additive/core/microstructure_3d.py

```diff
@@ -0,0 +1,397 @@
+        max_bulk_nucleation_density: int = DEFAULT_MAX_NUCLEATION_DENSITY_BULK,
+        num_initial_random_nuclei: int = DEFAULT_NUMBER_OF_RANDOM_NUCLEI,
+        machine: AdditiveMachine = AdditiveMachine(),
+        material: AdditiveMaterial = AdditiveMaterial(),
+    ):
+        """Initialize a ``Microstructure3DInput`` object."""
+
+        self.id = id
+        self.sample_min_x = sample_min_x
+        self.sample_min_y = sample_min_y
+        self.sample_min_z = sample_min_z
+        self.sample_size_x = sample_size_x
+        self.sample_size_y = sample_size_y
+        self.sample_size_z = sample_size_z
+        self.calculate_initial_microstructure = calculate_initial_microstructure
+        self.use_transient_bulk_nucleation = use_transient_bulk_nucleation
+        self.max_bulk_nucleation_density = max_bulk_nucleation_density
+        self.num_initial_random_nuclei = num_initial_random_nuclei
+        self.machine = machine
+        self.material = material
+
+    def __repr__(self):
+        repr = type(self).__name__ + "\n"
+        for k in self.__dict__:
+            if k == "_machine" or k == "_material":
+                repr += "\n" + k.replace("_", "", 1) + ": " + str(getattr(self, k))
+            else:
+                repr += k.replace("_", "", 1) + ": " + str(getattr(self, k)) + "\n"
+        return repr
+
+    def __eq__(self, __o: object) -> bool:
+        if not isinstance(__o, Microstructure3DInput):
+            return False
+        return (
+            self.id == __o.id
+            and self.sample_min_x == __o.sample_min_x
+            and self.sample_min_y == __o.sample_min_y
+            and self.sample_min_z == __o.sample_min_z
+            and self.sample_size_x == __o.sample_size_x
+            and self.sample_size_y == __o.sample_size_y
+            and self.sample_size_z == __o.sample_size_z
+            and self.calculate_initial_microstructure == __o.calculate_initial_microstructure
+            and self.use_transient_bulk_nucleation == __o.use_transient_bulk_nucleation
+            and self.max_bulk_nucleation_density == __o.max_bulk_nucleation_density
+            and self.num_initial_random_nuclei == __o.num_initial_random_nuclei
+            and self.machine == __o.machine
+            and self.material == __o.material
+        )
+
+    @staticmethod
+    def __validate_range(value, min, max, name):
+        if math.isnan(value):
+            raise ValueError("{} must be a number.".format(name))
+        if value < min or value > max:
+            raise ValueError("{} must be between {} and {}.".format(name, min, max))
+
+    @staticmethod
+    def __validate_greater_than_zero(value, name):
+        if math.isnan(value):
+            raise ValueError("{} must be a number.".format(name))
+        if value <= 0:
+            raise ValueError("{} must be greater than zero.".format(name))
+
+    @property
+    def id(self) -> str:
+        """User-provided ID for this simulation."""
```

Comment:
```suggestion
        """User-provided ID for the simulation."""
```
Given that we are operating on "self," we just use "the" rather than this. We started this because of the overuse of "this" in the doc rather than any real style rule.